pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/TopicSchema.java

```diff
@@ -158,6 +173,20 @@ private static SchemaType getDefaultSchemaType(Class<?> clazz) {
         }
     }
 
+    private static <T> AvroSchema<T> buildAvroSchema(Class<T> clazz, ConsumerConfig conf) {
+        String jsr310ConversionEnabled = null;
+        String alwaysAllowNull = null;
+        if (conf.getSchemaProperties() != null) {
```

Comment:
Please use `SchemaDefinitionBuilder#withProperties` to construct the schema definition and avoid adding this kind of customization code. Also, the schema properties should be passed to other schemas as well.

Reply:
> Please use `SchemaDefinitionBuilder#withProperties` to construct the schema definition and avoid adding this kind of customization code. Also, the schema properties should be passed to other schemas as well.

Modify as follows:
1 Since the map generated by protobuf is immutable, a new Map is added to TopicSchema line 161.
2 Since the parameters such as JSR310 in SchemaDefinitionBuilder are underlined,the incoming parameters will be converted